internal/nginx-meshctl/commands/services.go

```diff
@@ -29,60 +38,60 @@ func GetServices() *cobra.Command {
 
 	cmd.PersistentPreRunE = defaultPreRunFunc()
 	cmd.RunE = func(cmd *cobra.Command, args []string) error {
-		meshClient, err := mesh.NewMeshClient(initK8sClient.Config(), meshTimeout)
+		var meshServices []serviceDetails
+		k8sClient, err := client.New(initK8sClient.Config(), client.Options{})
 		if err != nil {
-			fmt.Println(genericGetServicesErrorInfo)
-
+			fmt.Printf("failed to initialize k8s client: %v\n", err)
 			return err
 		}
 		ctx, cancel := context.WithCancel(context.Background())
 		defer cancel()
 
-		res, err := meshClient.GetServices(ctx)
-		if err != nil {
-			fmt.Println(genericGetServicesErrorInfo)
-
+		services := &v1.ServiceList{}
+		if err := k8sClient.List(ctx, services); err != nil {
+			fmt.Printf("error getting list of services: %v\n", err)
 			return err
 		}
-		defer func() {
-			closeErr := res.Body.Close()
-			if closeErr != nil {
-				fmt.Println(closeErr)
-			}
-		}()
-
-		if res.StatusCode != http.StatusOK {
-			fmt.Println(genericGetServicesErrorInfo)
-
-			return mesh.ParseAPIError(res)
-		}
 
-		data, err := mesh.ParseGetServicesResponse(res)
-		if err != nil {
-			fmt.Println(genericGetServicesErrorInfo)
-
-			return fmt.Errorf("error parsing response body from NGINX Service Mesh API Server: %w", err)
+		for _, serviceObj := range services.Items {
+			serviceObj := serviceObj
+			// if we consider a service injectable then we consider it registered with our mesh
+			if injectable, err := isNamespaceInjectionEnabled(ctx, k8sClient, serviceObj.Namespace); err == nil && injectable {
+				upstreams, epErr := getEndpoints(ctx, k8sClient, serviceObj)
+				if err != nil {
+					return epErr
+				}
+				newDetails := serviceDetails{
+					name:      serviceObj.Name,
+					namespace: serviceObj.Namespace,
+					ports:     serviceObj.Spec.Ports,
+					addresses: upstreams,
+				}
+				meshServices = append(meshServices, newDetails)
```

Comment:
We can probably just create and append the new serviceDetails instance in one go.